DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/task_sequence_hardware_reuse/loop/src/main.cpp

```diff
@@ -0,0 +1,127 @@
+#include <iostream>
+
+// oneAPI headers
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+
+// Forward declare the kernel name in the global scope. This is an FPGA best
+// practice that reduces name mangling in the optimization reports.
+class VectorOpID;
+
+constexpr int kVectSize = 3;
+constexpr int N = 5;
+struct D3Vector {
+  float d[3];
+};
```

Comment:
```suggestion
using D3Vector = std::array<float, 3>;
```

I recommend you to use `std::array` for passing bundled data by value. This way you can poll the type for its length in loops. Instead of 

```c++
D3Vector vec = <...>;
for(int i = 0; i < 3; i++) {
  vec[i] = ...
}
```

You can do 

```c++
D3Vector vec = <...>;
for(int i = 0; i < D3Vector{}.size(); i++) {
  vec[i] = ...
}
```

or even

```c++
D3Vector vec = <...>;
for(int i = 0; i < vec.size(); i++) {
  vec[i] = ...
}
```

this is much safer code as you avoid possibly mismatching the loop bounds and the array size. it also makes your code clearer because the reader knows right away that you're iterating across all the elements of a D3Vector and not some magic number 😊.
This is equivalent to comparing with a constant because the `size()` function of `std::array` is `constepxr`: https://stackoverflow.com/questions/16866033/getting-the-number-of-elements-in-stdarray-at-compile-time